### PR TITLE
fix: Replace bare except with explicit RuntimeError handling (#101)

### LINKED ISSUE #101: Replace bare except with explicit exception in model loading (evaluate.py)
### Problem
In `evaluate.py`, the model loading section uses a bare `except:` clause:

```python
try:
    train_state.model.load_state_dict(torch.load(eval_cfg.checkpoint, map_location="cuda"), assign=True)
except:
    train_state.model.load_state_dict({k.removeprefix("_orig_mod."): v for k, v in torch.load(eval_cfg.checkpoint, map_location="cuda").items()}, assign=True)
```

Using a bare `except:` is dangerous because it catches all exceptions, including KeyboardInterrupt and SystemExit, which can hide bugs, interrupt signals, or errors that should be handled differently.

### Recommended Fix
Replace the bare `except:` with explicit exception(s) such as `except RuntimeError as e:` or another relevant exception. Also, consider printing/logging the exception for easier debugging.

### Proposed Solution
Example update:
```python
try:
    train_state.model.load_state_dict(torch.load(eval_cfg.checkpoint, map_location="cuda"), assign=True)
except RuntimeError as e:
    print(f"Model load error: {e}")
    train_state.model. load_state_dict({k. removeprefix("_orig_mod."): v for k, v in torch.load(eval_cfg. checkpoint, map_location="cuda").items()}, assign=True)
```

---
**Benefits:**
- Conforms to Python best practices
- Prevents masking other errors
- Improves maintainability and debugging

### References
- [PEP 8: Programming Recommendations](https://peps.python.org/pep-0008/#programming-recommendations)
- [Why you should never use bare except](https://docs.python.org/3/tutorial/errors.html#handling-exceptions)

### PR DESCRIPTION
**Pull request by somdipto**
## Summary
Fixed the bare `except:` clause in evaluate.py that was catching all exceptions including KeyboardInterrupt and SystemExit.

## Changes
- Replaced bare `except:` with explicit `RuntimeError` handling
- Added specific check for torch.compile prefix issues
- Load checkpoint once to avoid redundant I/O
- Re-raise other errors instead of silently hiding them

## Testing
- Works with normal checkpoints
- Works with torch.compile checkpoints
- Properly propagates other errors

## Linked Issue
Fixes #101

### CODE REVIEW

**Comment by somdipto** [comment]
hey guys can you please review my PR and leme know if there are any issue in that
